src/analyzer/analyzerbeats.cpp

```diff
@@ -13,20 +13,16 @@
 #include "track/beatutils.h"
 #include "track/track.h"
 
-// static
-QList<mixxx::AnalyzerPluginInfo> AnalyzerBeats::availablePlugins() {
+QList<mixxx::AnalyzerPluginInfo> defaultAvailableBeatsPlugins() {
```

Comment:
This is now a global free function without a namespace. I think In this case it is best to just manually inline it into  
AnalyzerBeats::availablePlugins() and use that one instead.

Reply:
Good eye and agreed on removing the global free-function. Now made the function namespaced.

---

Looking at the code design, while refactoring to summarise a simple common plugins interface, it seems that (the terms) "`availablePlugins`" and "`defaultPlugins`" have a different use-purpose.

To check to see what plugins are available: there is the supposedly ["availablePlugins" function API.](#diff-4186bd7b4da09d30a81e5b021b8991aacb26f9e75310c37403199da822c506beL60-L70)

To use the appropriate initial default plugins for the {Beats,Key}Plugin, "[defaultPlugins" is the API to use. ](https://github.com/mixxxdj/mixxx/blob/6bf5a0a94bdd1f1152f19ba30b25e29d7731b6e1/src/analyzer/analyzerkey.cpp#L16-L24)

From observations, it seems planned future-wise, if there are more "plugins", the "defaultPlugins" {Beats,Key} can differ depending on the compile-time settings, but the API to always get the "availablePlugins" must be somewhat stable, to "perform" the same API requirements regardless of whichever (future) plugin.

---

In the end, if the main blocking concern was the global free-function, then the rest of the PR should be fine and ok to merge these changes to mainline.